samples/zephyr-hello/build.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+BOARD_ARCH=arm_cortex_m3
+BOARD_NAME=qemu_cortex_m3
+export ZEPHYR_BASE="PLEASE_SET_ZEPHYR_BASE"
+
+if [ "$ZEPHYR_BASE" == "PLEASE_SET_ZEPHYR_BASE" ] ; then
+    echo "Please set ZEPHYR_BASE in this build.sh."
+    exit 1
+fi
+
+export KONAN_DATA_DIR=$HOME/.konan
+export KONAN_DEPS=$KONAN_DATA_DIR/dependencies
+
+DIR=$(cd "$(dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+if [ -z "$KONAN_HOME" ]; then
+    PATH="$DIR/../../dist/bin:$DIR/../../bin:$PATH"
+else
+    PATH="$KONAN_HOME/bin:$PATH"
+fi
+
+if [ x$TARGET == x ]; then
+case "$OSTYPE" in
+  darwin*)  TOOLCHAIN=gcc-arm-none-eabi-7-2017-q4-major-mac ;;
+  linux*)   TOOLCHAIN=gcc-arm-none-eabi-7-2017-q4-major-linux ;;
+  *)        echo "unknown: $OSTYPE" && exit 1;;
+esac
+fi
+
+GNU_ARM="$KONAN_DEPS/$TOOLCHAIN"
+
+rm -rf $DIR/build || exit 1
+mkdir -p $DIR/build && cd $DIR/build
+
+export ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb
+export GNUARMEMB_TOOLCHAIN_PATH=$GNU_ARM
+
+[ -f CMakeCache.txt ] || cmake -DCMAKE_VERBOSE_MAKEFILE=ON -DBOARD=$BOARD .. || exit 1
+
+# We need generated headers to be consumed by `cinterop`,
+# so we preconfigure the project with `make zephyr`.
+ninja zephyr
+
+rm -f program.o
+
+mkdir -p $DIR/build/kotlin
+
+konanc $DIR/src/main.kt \
+        -target zephyr_$BOARD_ARCH \
+        -opt -g -o $DIR/build/kotlin/program || exit 1
```

Comment:
Use either -opt or -g, better -opt.

Reply:
With -opt:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      200040 B       256 KB     76.31%
            SRAM:        8024 B        64 KB     12.24%
```
With -g:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      205400 B       256 KB     78.35%
            SRAM:        8032 B        64 KB     12.26%
```
So it saves 5 KBytes - does it worth a few more seconds to compile?

Reply:
And doing something not so trivial, for example:
``` Kotlin
fun main(args: Array<String>) {
    val a = 2.0
    val b = 2.0
    println("$a + $b = ${a + b}")
}
```
With `-opt` does not print `exit` at the end, so probably hangs somewhere.
With `-g` it works.

Reply:
Without flags it hangs too, only `-g` variant works.

Reply:
It shall be debugged, maybe it's something absolutely different from debug/no debug (i.e. certain image size).